### PR TITLE
feat: Use tag ids instead of values wherever possible

### PR DESCRIPTION
**Pull request by sradevski**
CLOSES CC-284

### CODE REVIEW

File: packages/admin-next/dashboard/src/hooks/table/query/use-product-table-query.tsx [hunk 1]
```diff
@@ -54,7 +54,7 @@ export const useProductTableQuery = ({
     collection_id: collection_id?.split(","),
     is_giftcard: is_giftcard ? is_giftcard === "true" : undefined,
     order: order,
-    tags: tags ? { value: tags.split(",") } : undefined,
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/hooks/table/query/use-product-table-query.tsx:57]: This was wrong, the tags here were IDs, not values

File: packages/core/core-flows/src/product/helpers/normalize-for-import.ts [hunk 2]
```diff
@@ -108,10 +116,15 @@ const normalizeProductForImport = (
     }
 
     if (normalizedKey.startsWith("product_tag_")) {
-      response["tags"] = [
-        ...(response["tags"] || []),
-        { value: normalizedValue },
-      ]
+      const tag = tagsMap.get(normalizedValue)
+      if (!tag) {
```
Comment (sradevski) [inline packages/core/core-flows/src/product/helpers/normalize-for-import.ts:120]: I would be ok for the import to first create all tags and then create the products if we want that behavior (as a separate PR though).

File: packages/admin-next/dashboard/src/hooks/table/query/use-product-table-query.tsx (hunk 1 above)
Comment (olivermrbl) [reply packages/admin-next/dashboard/src/hooks/table/query/use-product-table-query.tsx:57]: Nice catch

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**